File: src/test/fixtures/sample.go
// Package sample は計算機ユーティリティを提供する。
//
// テスト用フィクスチャファイル — Go サンプル
package sample

import (
	"errors"
	"fmt"
	"log"
	"time"
)

// ─── 型定義 ───

// HistoryEntry は計算履歴の1エントリを表す。
type HistoryEntry struct {
	Operation string
	Operands  [2]float64
	Result    float64
	Timestamp time.Time
}

// Calculator は簡単な計算機。
type Calculator struct {
	history      []HistoryEntry
	maxHistory   int
	enableLog    bool
	logger       *log.Logger
}

// ─── コンストラクタ ───

// NewCalculator は新しい Calculator を生成する。
func NewCalculator(maxHistory int, enableLog bool, logger *log.Logger) *Calculator {
	if logger == nil {
		logger = log.Default()
	}
	c := &Calculator{
		history:    make([]HistoryEntry, 0, maxHistory),
		maxHistory: maxHistory,
		enableLog:  enableLog,
		logger:     logger,
	}
	c.log("Calculator created (maxHistory=%d)", maxHistory)
	return c
}

// ─── 公開 API ───

// Add は2つの数を加算して返す。
func (c *Calculator) Add(a, b float64) float64 {
	result := a + b
	c.record("add", a, b, result)
	return result
}

// Multiply は掛け算を行う。
func (c *Calculator) Multiply(a, b float64) float64 {
	result := a * b
	c.record("multiply", a, b, result)
	return result
}

// Divide は割り算を行う。ゼロ除算時はエラーを返す。
func (c *Calculator) Divide(a, b float64) (float64, error) {
	if b == 0 {
		c.log("ERROR: division by zero: %.2f / %.2f", a, b)
		return 0, errors.New("cannot divide by zero")
	}
	result := a / b
	c.record("divide", a, b, result)
	return result, nil
}

// History は計算履歴のコピーを返す。
func (c *Calculator) History() []HistoryEntry {
	copied := make([]HistoryEntry, len(c.history))
	copy(copied, c.history)
	return copied
}

// ClearHistory は履歴をクリアする。
func (c *Calculator) ClearHistory() {
	c.history = c.history[:0]
	c.log("History cleared")
}

// ─── ヘルパー関数 ───

// Greet は挨拶メッセージを返す。
func Greet(name string) string {
	if name == "" {
		return "Hello, World!"
	}
	return fmt.Sprintf("Hello, %s!", name)
}

// ─── 内部メソッド ───

func (c *Calculator) record(op string, a, b, result float64) {
	entry := HistoryEntry{
		Operation: op,
		Operands:  [2]float64{a, b},
		Result:    result,
		Timestamp: time.Now(),
	}
	c.history = append(c.history, entry)
	if len(c.history) > c.maxHistory {
		c.history = c.history[1:]
	}
	c.log("%s(%.2f, %.2f) = %.2f", op, a, b, result)
}

func (c *Calculator) log(format string, args ...interface{}) {
	if c.enableLog {
		c.logger.Printf("[Calculator] "+format, args...)
	}
}

File: src/test/fixtures/sample.cs
/// <summary>
/// テスト用フィクスチャファイル — C# サンプル
///
/// 計算機ユーティリティ。クラス + インターフェース + 列挙型を含む。
/// </summary>

using System;
using System.Collections.Generic;
using Microsoft.Extensions.Logging;

namespace CodeWalker.Sample
{
    // ─── 列挙型 ───

    /// <summary>演算の種類</summary>
    public enum OperationType
    {
        Add,
        Multiply,
        Divide,
    }

    // ─── インターフェース ───

    /// <summary>計算機の公開契約</summary>
    public interface ICalculator
    {
        double Multiply(double a, double b);
        double Divide(double a, double b);
        IReadOnlyList<HistoryEntry> History { get; }
        void ClearHistory();
    }

    // ─── 履歴エントリ ───

    /// <summary>計算履歴の1エントリ</summary>
    public record HistoryEntry(
        OperationType Operation,
        double[] Operands,
        double Result,
        DateTime Timestamp);

    // ─── メインクラス ───

    /// <summary>
    /// 簡単な計算機クラス。
    /// 四則演算と履歴管理を提供する。
    /// </summary>
    public class Calculator : ICalculator
    {
        private readonly List<HistoryEntry> _history = new();
        private readonly int _maxHistory;
        private readonly ILogger<Calculator> _logger;

        public Calculator(int maxHistory = 100, ILogger<Calculator>? logger = null)
        {
            _maxHistory = maxHistory;
            _logger = logger ?? LoggerFactory
                .Create(b => b.AddConsole())
                .CreateLogger<Calculator>();
            _logger.LogDebug("Calculator instance created");
        }

        // --- 公開 API ---

        /// <summary>掛け算</summary>
        public double Multiply(double a, double b)
        {
            var result = a * b;
            Record(OperationType.Multiply, a, b, result);
            return result;
        }

        /// <summary>割り算（ゼロ除算チェック付き）</summary>
        public double Divide(double a, double b)
        {
            if (b == 0)
            {
                _logger.LogError("Division by zero: {A} / {B}", a, b);
                throw new DivideByZeroException("Cannot divide by zero");
            }
            var result = a / b;
            Record(OperationType.Divide, a, b, result);
            return result;
        }

        /// <summary>履歴を取得</summary>
        public IReadOnlyList<HistoryEntry> History => _history.AsReadOnly();

        /// <summary>履歴をクリア</summary>
        public void ClearHistory()
        {
            _history.Clear();
            _logger.LogInformation("History cleared");
        }

        // --- 内部ヘルパー ---

        private void Record(OperationType op, double a, double b, double result)
        {
            var entry = new HistoryEntry(op, new[] { a, b }, result, DateTime.UtcNow);
            _history.Add(entry);
            if (_history.Count > _maxHistory)
            {
                _history.RemoveAt(0);
            }
            _logger.LogDebug("{Op}({A}, {B}) = {Result}", op, a, b, result);
        }
    }

    // ─── ヘルパー関数 ───

    /// <summary>挨拶ユーティリティ</summary>
    public static class Greeter
    {
        public static string Greet(string? name)
        {
            if (string.IsNullOrEmpty(name))
            {
                return "Hello, World!";
            }
            return $"Hello, {name}!";
        }
    }
}
